packages/http-server/src/http-server.ts

```diff
@@ -29,18 +33,27 @@ type CallbackMap = {
   getExchange?: GetExchangeCallback
   getExchanges?: GetExchangesCallback
   getOfferings?: GetOfferingsCallback
+  getBalances?: GetBalancesCallback
   createExchange?: CreateExchangeCallback
   submitOrder?: SubmitOrderCallback
   submitClose?: SubmitCloseCallback
 }
 
 /**
  * Options for creating a new HttpServer
+ * @param opts.offeringsApi Optionally accepts an {@link OfferingsApi}. Defaults to an {@link InMemoryOfferingsApi} which supports additional methods.
+ * @param opts.exchangesApi Optionally accepts an {@link ExchangesApi}. Defaults to an {@link InMemoryExchangesApi} which supports additional methods.
+ * @param opts.balances Optionally accepts an `{ }` with 0-1 properties to indicate that this HttpServer should support the optional `balance` endpoint. Else, leave `undefined`.
+ * @param opts.balancesApi Optionally accepts a {@link BalancesApi}. Defaults to an {@link InMemoryBalancesApi} which supports additional methods.
+ * @param opts.pfiDid Required if instantiating the HttpServer with options. Else, defaults to an arbitrary string for example purposes only.
  * @beta
  */
 type NewHttpServerOptions = {
   offeringsApi?: OfferingsApi
   exchangesApi?: ExchangesApi,
+  balances?: {
+    balancesApi?: BalancesApi,
+  }
```

Comment:
why not? (like the others)

```suggestion
  balancesApi?: BalancesApi,
```

Reply:
is this relevant to the line of code in the constructor?

```ts
this.balancesApi = opts?.balances ? opts?.balances.balancesApi ?? new InMemoryBalancesApi() : undefined
```

so the idea being PFIs may choose to not support the balances endpoint at all yeah? in that case you could make the default (of my proposal) `undefined` instead of `InMemoryBalancesApi`

Reply:
gotcha so if the developer wants to implement the balances endpoint and wants to use the `InMemoryBalancesApi` then they would write

```ts
new TbdexHttpServer({ balances: { }, pfiDid: 'did:ex:pfi' })
```

---

but in this case, the developer knows they want to implement it so is that better than this? (assume the default of `NewHttpServerOptions.balancesApi` is `undefined` instead of the in memory instance)

```ts
new TbdexHttpServer({ balancesApi: InMemoryBalancesApi(), pfiDid: 'did:ex:pfi' })
```